src/OrleansRuntime/GrainTypeManager/GrainTypeData.cs

```diff
@@ -151,5 +154,11 @@ internal static PlacementStrategy GetPlacementStrategy(Type grainClass, Placemen
 
             return predicate;
         }
+
+        private ParameterInfo[] GetConstructorParameters()
+        {
+            var constructor = this.Type.GetConstructors(BindingFlags.Public | BindingFlags.Instance).FirstOrDefault();
+            return constructor?.GetParameters() ?? EmptyParameters;
+        }
```

Comment:
I'd strongly suggest that we don't couple GrainTypeData to this. The reason is that this caching optimization (because it's nothing more than that) made it to something more core to Orleans, and IMO still needs more thought with regards to 3rd party containers. Other containers (or even other implementations of `IGrainActivator` might select a different constructor. I think if there's a need to couple it to something, I'd couple it to `DefaultGrainActivator`. Otherwise, probably even more flexible, make it completely standalone so that if you replace that other logic, you could replace this too.